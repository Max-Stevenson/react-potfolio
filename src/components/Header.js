import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

export const Header = () => (
	<header className="header">
		<div className="content-container">
			<div className="header__content">
				<h1>Portfolio</h1>
				<NavLink className="is_active" exact={true} to="/">Home</NavLink>
				<NavLink className="is_active" exact={true} to="/portfolio">Portfolio</NavLink>
				<NavLink className="is_active" to="/contact">Contact</NavLink>
			</div>
		</div>
	</header>
);

const mapDispatchToProps = (dispatch) => ({
	startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);