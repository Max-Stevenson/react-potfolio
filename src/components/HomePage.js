import React from 'react';

const HomePage = () => (
  <div>
    <h1>My Portfolio</h1>
    <p>Welcome to my site. Take a look around!</p>
  </div>
);

export default HomePage;