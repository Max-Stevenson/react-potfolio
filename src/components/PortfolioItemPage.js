import React from 'react';

const HomePage = (props) => (
  <div>
    <h1>A Portfolio Item</h1>
    <p>This is Portfolio item: {props.match.params.id}</p>
  </div>
);

export default HomePage;