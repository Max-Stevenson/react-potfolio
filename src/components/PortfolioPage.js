import React from 'react';
import { Link } from 'react-router-dom';

const HomePage = (props) => (
  <div>
    <h1>My Work</h1>
    <p>Checkout some of my projects:</p>
    <Link to="/portfolio/1">Item One</Link>
    <Link to="/portfolio/2">Item Two</Link>
  </div>
);

export default HomePage;