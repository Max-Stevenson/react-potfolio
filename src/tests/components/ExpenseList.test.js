import { ExpenseList } from '../../components/ExpenseList';
import { shallow } from 'enzyme';
import React from 'react';
import expenses from '../fixtures/expenses';

test('Should render Expense List with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses}/>);
  expect(wrapper).toMatchSnapshot();
});