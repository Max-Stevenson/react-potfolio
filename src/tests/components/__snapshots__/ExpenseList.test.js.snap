// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense List with expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenes
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={1}
      createdAt={0}
      description="test"
      id="1"
      key="1"
      note="test"
    />
    <ExpenseListItem
      amount={1}
      createdAt={-345600000}
      description="test two"
      id="2"
      key="2"
      note="test"
    />
    <ExpenseListItem
      amount={3}
      createdAt={345600000}
      description="test three"
      id="3"
      key="3"
      note="test"
    />
  </div>
</div>
`;
